SU2_CFD/src/solver_adjoint_discrete.cpp

```diff
@@ -1041,39 +1094,39 @@ void CDiscAdjSolver::LoadRestart(CGeometry **geometry, CSolver ***solver, CConfi
 
 }
 
-void CDiscAdjSolver::ComputeResidual_Multizone(CGeometry *geometry, CConfig *config){
+//void CDiscAdjSolver::ComputeResidual_Multizone(CGeometry *geometry, CConfig *config){
 
-  unsigned short iVar;
-  unsigned long iPoint;
-  su2double residual, bgs_sol;
+//  unsigned short iVar;
+//  unsigned long iPoint;
+//  su2double residual, bgs_sol;
 
-  /*--- Set Residuals to zero ---*/
+//  /*--- Set Residuals to zero ---*/
 
-  for (iVar = 0; iVar < nVar; iVar++){
-      SetRes_BGS(iVar,0.0);
-      SetRes_Max_BGS(iVar,0.0,0);
-  }
+//  for (iVar = 0; iVar < nVar; iVar++){
+//      SetRes_BGS(iVar,0.0);
+//      SetRes_Max_BGS(iVar,0.0,0);
+//  }
 
-  /*--- Compute the BGS solution (adding the cross term) ---*/
-  for (iPoint = 0; iPoint < nPointDomain; iPoint++){
-    for (iVar = 0; iVar < nVar; iVar++){
-      bgs_sol = node[iPoint]->GetSolution(iVar) + node[iPoint]->GetCross_Term_Derivative(iVar);
-      node[iPoint]->Set_BGSSolution(iVar, bgs_sol);
-    }
-  }
+//  /*--- Compute the BGS solution (adding the cross term) ---*/
+//  for (iPoint = 0; iPoint < nPointDomain; iPoint++){
+//    for (iVar = 0; iVar < nVar; iVar++){
+//      bgs_sol = node[iPoint]->GetSolution(iVar) + node[iPoint]->GetCross_Term_Derivative(iVar);
+//      node[iPoint]->Set_BGSSolution(iVar, bgs_sol);
+//    }
+//  }
 
-  /*--- Set the residuals ---*/
+//  /*--- Set the residuals ---*/
 
-  for (iPoint = 0; iPoint < nPointDomain; iPoint++){
-      for (iVar = 0; iVar < nVar; iVar++){
-          residual = node[iPoint]->Get_BGSSolution(iVar) - node[iPoint]->Get_BGSSolution_k(iVar);
+//  for (iPoint = 0; iPoint < nPointDomain; iPoint++){
+//      for (iVar = 0; iVar < nVar; iVar++){
+//          residual = node[iPoint]->Get_BGSSolution(iVar) - node[iPoint]->Get_BGSSolution_k(iVar);
 
-          AddRes_BGS(iVar,residual*residual);
-          AddRes_Max_BGS(iVar,fabs(residual),geometry->node[iPoint]->GetGlobalIndex(),geometry->node[iPoint]->GetCoord());
-      }
-  }
+//          AddRes_BGS(iVar,residual*residual);
```

Comment:
same comment about commented out code

Reply:
Oh! That commented out code is an issue.
I commented it out because of its design flaw to compute solutions where it should only compute residuals (which caused some problems). So I used the base class implementation this way.
I'll revert it and add `if` which I don't like at all but I think there's no way around.

Reply:
But if the code works with this commented out, and it has a design flaw, I would say remove it entirely.